burn-core/src/record/serde/data.rs

```diff
@@ -0,0 +1,261 @@
+        match self {
+            NestedValue::String(string) => Some(string),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as a f32.
+    pub fn as_f32(self) -> Option<f32> {
+        match self {
+            NestedValue::F32(f32) => Some(f32),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as a f64.
+    pub fn as_f64(self) -> Option<f64> {
+        match self {
+            NestedValue::F64(f64) => Some(f64),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as an i16.
+    pub fn as_i16(self) -> Option<i16> {
+        match self {
+            NestedValue::I16(i16) => Some(i16),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as an i32.
+    pub fn as_i32(self) -> Option<i32> {
+        match self {
+            NestedValue::I32(i32) => Some(i32),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as an i64.
+    pub fn as_i64(self) -> Option<i64> {
+        match self {
+            NestedValue::I64(i64) => Some(i64),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as a u16.
+    pub fn as_u16(self) -> Option<u16> {
+        match self {
+            NestedValue::U16(u16) => Some(u16),
+            _ => None,
+        }
+    }
+
+    /// Get the nested value as a u64.
+    pub fn as_u64(self) -> Option<u64> {
+        match self {
+            NestedValue::U64(u64) => Some(u64),
+            _ => None,
+        }
+    }
+
+    /// Deserialize a nested value into a record type.
+    pub fn try_into_record<T, PS, A, B>(self, device: &B::Device) -> Result<T, Error>
+    where
+        B: Backend,
+        T: Record<B>,
+        PS: PrecisionSettings,
+        A: BurnModuleAdapter,
+    {
+        let deserializer = Deserializer::<A>::new(self, false);
+
+        let item = T::Item::deserialize(deserializer)?;
+
+        // Convert the deserialized item into a Record instance
+        Ok(T::from_item::<PS>(item, device))
+    }
+}
+
+/// Remap the tensor locations according to the key remapping.
+///
+/// # Arguments
+///
+/// * `tensors` - A map of tensors.
+/// * `key_remap` - A vector of tuples containing a regular expression and a replacement string.
+///                See [regex::Regex::replace](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace)
+///                for more information.
+///
+/// # Returns
+///
+/// A map of tensors with the remapped keys.
+pub fn remap<T>(
+    mut tensors: HashMap<String, T>,
+    key_remap: Vec<(Regex, String)>,
+) -> HashMap<String, T> {
+    if key_remap.is_empty() {
+        return tensors;
+    }
+
+    let mut remapped = HashMap::new();
+
+    for (name, tensor) in tensors.drain() {
+        let mut new_name = name.clone();
+        for (pattern, replacement) in &key_remap {
+            if pattern.is_match(&name) {
+                new_name = pattern.replace_all(&name, replacement.as_str()).to_string();
```

Comment:
Since `LoadArgs::with_key_remap` inserts `^` and `$`, `replace_all` has no effect at all here.